### DISCUSSION TITLE
External EXTENSIONS_PATH possible?

### DISCUSSION THREAD

**Discussion by torstenlemke** [directus/directus#10208] [Archive]
Hello 👋🏻 

Is it possible to outsource the extensions' directory?
I'm trying to modify the docker image to having zero needed volumes in terms of being able to allow zero downtime deployments.

In the docs it says the EXTENSIONS_PATH has to be a "Path to your local extensions folder.", but is there any way around this?

Cheers
Torsten

**Comment by rijkvanzanten** [comment]
Heya! There's no way around that. The extensions are loaded in through Node's module resolution (eg `require()`) so the file path needs to be accessible. 

> I'm trying to modify the docker image to having zero needed volumes in terms of being able to allow zero downtime deployments.

Out of curiosity, how are the volumes preventing you from doing zero downtime deployments? 

An alternative option would be to make a new Dockerfile yourself that uses `directus/directus` as the FROM. That way you can bundle the extensions on the container image instead of having to rely on a "live" volume 🙂 

Something like:

```Dockerfile
ARG DIRECTUS_VERSION
FROM directus/directus:${DIRECTUS_VERSION}
ENV NODE_ENV production
WORKDIR /directus
COPY ./extensions ./extensions
```

(the above assumes your local `./extensions` folder is pre-bundled and ready for launch)

**Reply by torstenlemke** [reply]
thank you for that answer!
well, the reason I was thinking about removing that, was the fact of some services like sloppy, prohibiting zero downtime deployments, if there's a volume connected.

I'll try that with the FROM solution above :)
Thanks
